src/induction/base.ml

```diff
@@ -295,16 +356,36 @@ let rec next (input_sys, aparam, trans, solver, k, unknowns) =
     else
      (* Looping. *)
      next
-      (input_sys, aparam, trans, solver, k_p_1, unfalsifiable)
+      (input_sys, aparam, trans, solver, k_p_1, out_of_bounds @ unfalsifiable, invs)
+
+and skip_steps_init trans solver =
+  let num_skip = lowest_lower_bound trans in
+  let step = ref Numeral.one in
+  while Numeral.to_int !step <= num_skip do
+    (* Declaring unrolled vars at k+1. *)
+    TransSys.declare_vars_of_bounds
+    trans (SMTSolver.declare_fun solver) !step !step ;
+
+    (* Asserting transition relation for next iteration. *)
+    TransSys.trans_of_bound (Some (SMTSolver.declare_fun solver)) trans !step
+    |> SMTSolver.assert_term solver ;
+
+    step := Numeral.succ !step;
+  done;
+  num_skip
 
 (* Initializes the solver for the first check. *)
-let init input_sys aparam trans =
+let init input_sys aparam trans invs =
```

Comment:
Rename the new bool argument `invs` with `no_skip` (in all functions). The current name is misleading.